server/vertx-server/src/main/scala/sttp/tapir/server/vertx/streams/Pipe.scala

```diff
@@ -156,10 +157,12 @@ object Pipe {
         case Pong(payload) =>
           writeFrame(VertxWebSocketFrame.pongFrame(Buffer.buffer(payload)))
         case Close(statusCode, _) =>
-          socket.reject(statusCode)
+          socket.close(statusCode.toShort, new Handler[AsyncResult[Void]] {
+            def handle(x: AsyncResult[Void]): Unit = ()
+          })
       }
 
-      if (socket.writeQueueFull()) {
+      if (!socket.isClosed && socket.writeQueueFull()) {
```

Comment:
[writeQueueFull](https://github.com/eclipse-vertx/vert.x/blob/462e3ea5203dd710e52a84bf8fba5f5a5dc5eb46/src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java#L133-L138) is checking and [throw an error](https://github.com/eclipse-vertx/vert.x/blob/462e3ea5203dd710e52a84bf8fba5f5a5dc5eb46/src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java#L441-L445) if the websocket is already closed.

Reply:
This is related to the above note on how we close the socket